language/src/ceylon/language/ArraySequence.ceylon

```diff
@@ -3,127 +3,106 @@
  Since [[Array]]s are mutable, this class is private to the
  language module, where we can be sure the `Array` is not
  modified after the `ArraySequence` has been initialized."
-by ("Tom")
+by("Tom")
 shared sealed final
 serializable
 tagged("Collections", "Sequences")
-class ArraySequence<out Element>(array)
+class ArraySequence<out Element>(Array<Element> array)
         extends Object()
         satisfies [Element+] {
     
-    Array<Element> array;
+    assert(!array.empty);
     
-    assert (!array.empty);
+    getFromFirst = array.getFromFirst;
     
-    getFromFirst(Integer index) 
-            => array.getFromFirst(index);
-    
-    contains(Object element) 
-            => array.contains(element);
+    contains = array.contains;
     
     size => array.size;
     
-    iterator() => array.iterator();
+    iterator => array.iterator;
     
     shared actual Element first {
-        if (exists first = array.first) {
-            return first;
-        }
-        else {
-            assert (is Element null);
-            return null;
-        }
+        assert(is Element first = array.first);
+        return first;
     }
     
     shared actual Element last {
-        if (exists last = array.last) {
-            return last;
-        }
-        else {
-            assert (is Element null);
-            return null;
-        }
+        assert(is Element last = array.last);
+        return last;
```

Comment:
Wait. What about `assert(exists last = array.last);`? `array.last` and `array.first` are already of type `Element?`, since `array` is of type `Array<Element>`.